src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorIntegrationTest.java

```diff
@@ -1,342 +1,369 @@
+
+            // check the ID field has the correct ID appended if necessary
+            if (inDbSNP) {
+                Assert.assertTrue(!inputVC.hasID() || outputVC.getID().startsWith(inputVC.getID()));
+                Assert.assertEquals(outputVC.getID().length() > inputVC.getID().length(), inDbSNPWithDifferentKey);
+            }
+        }
     }
 
+    // test the -resource argument and associated ID, FILTER and INFO field annotation
     @Test
-    public void testDBTagWithTwoComps() throws IOException {
-        assertVariantContextsMatch(getTestFile("vcfexample3empty.vcf"),
-                getTestFile("expected/testDBTagWithTwoComps.vcf"),
-                Arrays.asList("-G", "Standard", "-L", getToolTestDataDir() + "vcfexample3empty.vcf", "--comp:H3", getToolTestDataDir() + "fakeHM3.vcf", "--comp:foo", getToolTestDataDir() + "fakeHM3.vcf"),
-                b37_reference_20_21, Collections.emptyList());
-    }
+    public void testResource() {
+        final File inputVCF = BASIC_INPUT;
+        final File outputVCF = createTempFile("output", ".vcf");
+        final File resourceVcf = INDELS;
+
+        final Map<Integer, VariantContext> resourceVariantsByStart = VariantContextTestUtils.streamVcf(resourceVcf)
+                .collect(Collectors.toMap(VariantContext::getStart, vc -> vc));
+
+        final Map<String, VariantContext> resourceVariantsByStartAndAlleles = VariantContextTestUtils.streamVcf(resourceVcf)
+                .collect(Collectors.toMap(VariantAnnotatorIntegrationTest::keyForVariant, vc -> vc));
+
+        final String[] baseArgs = new ArgumentsBuilder()
+                .addVCF(inputVCF)
+                .addOutput(outputVCF)
+                .addFileArgument(StandardArgumentDefinitions.RESOURCE_LONG_NAME + ":" + FOO, resourceVcf)
+                .getArgsArray();
+
+        // test the --expression foo.FILTER annotation
+        final ArgumentsBuilder filterArgs = new ArgumentsBuilder(baseArgs).addArgument(VariantAnnotator.EXPRESSION_LONG_NAME, FOO_FILTER);
+        runCommandLine(filterArgs.getArgsList());
+        for (final VariantContext outputVC : VariantContextTestUtils.getVariantContexts(outputVCF)) {
+            final VariantContext resourceVariant = resourceVariantsByStart.get(outputVC.getStart());
+            if (resourceVariant != null) {
+                Assert.assertTrue(outputVC.hasAttribute(FOO_FILTER));
+                Assert.assertEquals(outputVC.getAttribute(FOO_FILTER), resourceVariant.isNotFiltered() ? "PASS" : String.join(";", new ArrayList<>(resourceVariant.getFilters())));
+            } else {
+                Assert.assertFalse(outputVC.hasAttribute(FOO_FILTER));
+            }
+        }
 
-    @Test
-    public void testNoQuals() throws IOException {
-        // NOTE, this test is asserting that the QD calculation is dependant on existing QUAL field, the values themselves are subject to random jitter
```

Comment:
That's not fair, because we can calculate QUAL from genotypes if they're present.  I agree we don't need to preserve this test.